File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        //Optimized Approach: Using min_heap: Time:O(nlogk) and Space:O(n)
        priority_queue<int,vector<int>,greater<int>> min_heap;
        
        for(auto it:nums)
        {
            min_heap.push(it);
            if(min_heap.size()>k){
                min_heap.pop();
            }
        }
        return min_heap.top();
    }
};

File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        PriorityQueue<Integer> min_heap = new PriorityQueue<Integer>();
        for(int i:nums)
        {
            min_heap.add(i);
            if(min_heap.size()>k)
                min_heap.remove();
        }
        return min_heap.peek();
    }
}